models/User.js

```diff
@@ -26,9 +26,13 @@ const UserSchema = new Schema({
 		unique: true,
 		dropDups: true
 	},
-	isAdmin: {
-		type: Boolean,
-		default: false
+	role: {
+		//Bitmap of permissions the user has
+		//For example, role=1 is a don, role=3 is a don and admin
+		//bit 0 = don
+		//bit 1 = admin
+		type: Number,
```

Comment:
Hmm, maybe use string for this? Doing `user.role === 'don'` is a lot easier to read than the `user.role&1 != 1` you have above. I'm not too worried about efficieny as we're not doing it too often, and I don't want people to have to check the schema every time they read a piece of code using a user's role.

Reply:
But what is a don is an admin? I know this is a very C thing to do, but I couldn't think of a better way to be able to give a user multiple roles. It could be user.role ==='don&admin&...', is that cleaner than a bitmap?

Reply:
Fair point. How about an array then? Then you'd do `if(user.role.includes('don') && user.role.includes('admin'))` which still is very easy to read without needing a comment explaining what it does.